frontend/src/selectors/collectives.js

```diff
@@ -0,0 +1,106 @@
+
+import { getExpensesSelector } from './expenses';
+import { getTransactionsSelector } from './transactions';
+
+import roles from '../constants/roles';
+
+const DEFAULT_COLLECTIVE_SETTINGS = {
+  lang: 'en',
+  formatCurrency: {
+    compact: false,
+    precision: 2
+  }
+};
+
+/*
+ * Collective selectors
+ */
+const getCollectivesSelector = (state) => state.collectives;
+
+export const getCollectiveSelector = createSelector(
+  [ getSlugSelector, getCollectivesSelector ],
+  (slug, collectives) => collectives[slug]);
+
+export const getCollectiveSettingsSelector = createSelector(
+  getCollectiveSelector,
+  (collective) => collective.settings || DEFAULT_COLLECTIVE_SETTINGS);
+
+export const getStripeAccountSelector = createSelector(
+  getCollectiveSelector,
+  (collective) => collective.stripeAccount || {});
+
+export const getStripePublishableKeySelector = createSelector(
+  getStripeAccountSelector,
+  (stripeAccount) => stripeAccount.stripePublishableKey);
+
+const getCollectiveUsersByRoleSelector = createSelector(
+  getCollectiveSelector,
+  (collective) => collective.usersByRole || {});
+
+export const getCollectiveHostSelector = createSelector(
+  getCollectiveUsersByRoleSelector,
+  (usersByRole) => usersByRole[roles.HOST] || []);
+
+export const getCollectiveMembersSelector = createSelector(
+  getCollectiveUsersByRoleSelector,
+  (usersByRole) => usersByRole[roles.MEMBER] || []);
+
+export const getCollectiveBackersSelector = createSelector(
+  getCollectiveUsersByRoleSelector,
+  (usersByRole) => usersByRole[roles.BACKER] || []);
+
+export const hasHostSelector = createSelector(
+  [getCollectiveHostSelector, getStripePublishableKeySelector],
+  (host, publishableKey) => host.length === 1 && publishableKey ? true : false);
+
+export const getCollectiveDataSelector = createSelector(
+  getCollectiveSelector,
+  (collective) => collective.data || {});
+
+export const getCollectiveContributorsSelector = createSelector(
+  getCollectiveDataSelector,
+  (data) => data.githubContributors ? formatGithubContributors(data.githubContributors) : []);
+
+export const getPopulatedCollectiveSelector = createSelector(
+  [ getCollectiveSelector,
+    getCollectiveSettingsSelector,
+    getCollectiveHostSelector,
+    getCollectiveMembersSelector,
+    getCollectiveBackersSelector,
+    getCollectiveContributorsSelector,
+    getExpensesSelector,
+    getTransactionsSelector ],
+    (collective, settings, host, members, backers, contributors, expenses, transactions) =>
+      Object.assign(
+        {},
+        collective,
+        settings,
+        { host },
+        { members },
+        { backers },
+        { backersCount: backers.length },
+        { contributors },
+        { contributorsCount: contributors.length },
+        { expenses: filterCollection(expenses, { GroupId: collective.id })},
+        { transactions: filterCollection(transactions, {GroupId: collective.id })},
+    ));
```

Comment:
I have mixed feeling with those selectors. I understand their utility but it's adding complexity. Is it worth it for us at our stage?

I'm all for giving it a try, but you'll have to show me more next week.